anta/tests/ptp2.py

```diff
@@ -0,0 +1,161 @@
+        ptp_mode = command_output["ptpMode"] if "ptpMode" in command_output.keys() else ""
+        if ptp_mode == "ptpBoundaryClock":
+            self.result.is_success(f"Valid PTP mode found: '{ptp_mode}'")
+        else:
+            self.result.is_failure(f"Device is not configured as a Boundary Clock: '{ptp_mode}'")
+
+
+class PtpGMStatus(AntaTest):
+    """
+    This test verifies that the device is locked to a valid GM
+    The user should provide a single "validGM" as an input
+    To test PTP failover, re-run the test with secondary GMid configured.
+    """
+
+    class Input(AntaTest.Input):  # pylint: disable=missing-class-docstring
+        validGM: str
+        """ expected validGM as input from the user in nrfu file"""
+
+    name = "PtpGMStatus"
+    description = "Check device is locked to an allowed GM"
+    categories = ["hardware"]
+    commands = [AntaCommand(command="show ptp", ofmt="json")]
+
+    # Verify that all switches are locked to the same GMID, and that this GMID is one of the provided GMs
+
+    @skip_on_platforms(["cEOSLab", "vEOS-lab"])
+    @AntaTest.anta_test
+    def test(self) -> None:
+        command_output = self.instance_commands[0].json_output
+        validGM = self.inputs.validGM
+        ptp_gmid = command_output["ptpClockSummary"]["gmClockIdentity"] if "gmClockIdentity" in command_output["ptpClockSummary"].keys() else ""
+        # print(command_output["ptpClockSummary"]["gmClockIdentity"])
+
+        if ptp_gmid == validGM:
+            self.result.is_success(f"Valid GM found: '{ptp_gmid}'")
+        else:
+            self.result.is_failure(f"Device is not locked to valid GM: '{ptp_gmid}'")
+
+
+class PtpLockStatus(AntaTest):
+    """
+    This test verifies that the device as a recent PTP lock
+    """
+
+    name = "PtpLockStatus"
+    description = "Check that the device was locked to the upstream GM in the last minute"
+    categories = ["hardware"]
+    commands = [AntaCommand(command="show ptp", ofmt="json")]
+
+    # Verify that last lock time is within the last minute
+
+    @skip_on_platforms(["cEOSLab", "vEOS-lab"])
+    @AntaTest.anta_test
+    def test(self) -> None:
+        command_output = self.instance_commands[0].json_output
+        ptp_lastSyncTime = command_output["ptpClockSummary"]["lastSyncTime"] if "lastSyncTime" in command_output["ptpClockSummary"].keys() else ""
+        ptp_currentPtpSystemTime = (
+            command_output["ptpClockSummary"]["currentPtpSystemTime"] if "currentPtpSystemTime" in command_output["ptpClockSummary"].keys() else ""
+        )
+        # print(command_output["ptpClockSummary"])
+
+        time_to_last_sync = ptp_currentPtpSystemTime - ptp_lastSyncTime
+
+        if time_to_last_sync <= 60:
+            self.result.is_success(f"Current PTP lock found: '{time_to_last_sync}'s")
+        else:
+            self.result.is_failure(f"Device Lock is old: '{time_to_last_sync}'s")
+
+
+class PtpOmfStatus(AntaTest):
```

Comment:
I assume the test name is `PtpOfmStatus` and not `PtpOmfStatus` ?
What do you think about `PtpOffset` or `PtpOffsetStatus` ?
Not sure if this is relevant for some testing scenarios, but you could expose the 1000ns value as input with a default value.

Reply:
using Ptpoffset